test/unit/radio.js

```diff
@@ -86,4 +90,44 @@ describe('MaterialRadio', function () {
     }())).to.equal('mdl-radio mdl-js-radio is-upgraded is-checked');
   });
 
+  it('should update related radios on one changing', function() {
+    var radios = [];
+    radios.push(createRadioWithValues('test', 'one'));
+    radios.push(createRadioWithValues('test', 'two'));
+    radios.push(createRadioWithValues('test', 'three'));
+    radios.push(createRadioWithValues('tester', 'A'));
+    radios.push(createRadioWithValues('tester', 'B'));
+    radios.push(createRadioWithValues('tester', 'C'));
+    var container = document.createElement('div');
+    radios.forEach(function(item) {
+      container.appendChild(item);
+      componentHandler.upgradeElement(item);
+    });
+    document.body.appendChild(container);
+
+    // Prepare the change event for manual firing.
+    // Used to trigger sibling checking as if UX triggered.
+    var changeEvent = document.createEvent("HTMLEvents");
+    changeEvent.initEvent("change", false, true);
+
+    // Check that all inputs are in a clean state
+    Array.prototype.splice.call(document.querySelectorAll('[type="radio"]')).forEach(function(item) {
+      expect(item.className).to.equal('mdl-radio mdl-js-radio is-upgraded');
+    });
+
+    radios[0].MaterialRadio.check();
+    radios[0].querySelector('input').dispatchEvent(changeEvent);
+    expect(radios[0].className).to.equal('mdl-radio mdl-js-radio is-upgraded is-checked');
+
+    radios[1].MaterialRadio.check();
+    radios[1].querySelector('input').dispatchEvent(changeEvent);
+    expect(radios[1].className).to.equal('mdl-radio mdl-js-radio is-upgraded is-checked');
+    expect(radios[0].className).to.equal('mdl-radio mdl-js-radio is-upgraded');
+
+    Array.prototype.splice.call(document.querySelectorAll('[type="radio"][name="tester"]')).forEach(function(item) {
+      expect(item.className).to.equal('mdl-radio mdl-js-radio is-upgraded');
+    });
```

Comment:
Yup, that's it. Just a buffer space to verify that things named roughly the same aren't touched by the onChange query. Comment incoming.